numpy/tests/test_public_api.py

```diff
@@ -287,10 +287,10 @@ def is_unexpected(name):
     return True
 
 
-if sys.version_info < (3, 12):
-    SKIP_LIST = ["numpy.distutils.msvc9compiler"]
-else:
+if sys.version_info >= (3, 12):
     SKIP_LIST = []
+else:
+    SKIP_LIST = ["numpy.distutils.msvc9compiler"]
```

Comment:
Not sure what the reasoning was for this one.

Reply:
See [PYI066](https://docs.astral.sh/ruff/rules/bad-version-info-order/#bad-version-info-order-pyi066):
> As a convention, branches that correspond to newer Python versions should come first. This makes it easier to understand the desired behavior, which typically corresponds to the latest Python versions.
> 
> This rule enforces the convention by checking for `if` tests that compare `sys.version_info` with `<` rather than `>=`.